src/addr.rs

```diff
@@ -158,12 +158,17 @@ impl VirtAddr {
     /// Aligns the virtual address upwards to the given alignment.
     ///
     /// See the `align_up` function for more information.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the resulting address is higher than
+    /// `0xffff_ffff_ffff_ffff`.
```

Comment:
Will this actually happen? Looking at the implementation it seems like this would just wrap to `VirtAddr(0)`

Reply:
This would have panicked in debug mode.

Now I changed the `align_up` to explicitly check for overflow even in release mode. Given that this was already panicking in debug mode, I'd argue that this isn't a breaking change.

Reply:
Oh that makes sense. Fix seems reasonable.